### PR TITLE
fix: correctly set nulls in string columns

### PR DESCRIPTION
**Pull request by alamb**
Closes https://github.com/influxdata/influxdb_iox/issues/3935

# Rationale

Re https://github.com/influxdata/conductor/issues/892

`NULL` strikes again!

The validity bit in String fields was not being correctly set. Thus string columns that were supposed to be `NULL` were actually seen as `""`, a subtle but critical difference for correctness!

This bug was likely masked for string columns specifically because most of our tests compare batches by printing them to strings where `""` and `null` are rendered the same.

Change:
1. Properly use validity mask
2. explicit tests for null when inserting null line protocol

I also verified this now passes the fluxtest suite:

```shell
isTypeTableString...success
```

### CODE REVIEW

File: mutable_batch/src/column.rs [hunk 1]
```diff
@@ -298,7 +298,7 @@ impl Column {
                     .context(CreatingArrowArraySnafu)?;
                 Arc::new(UInt64Array::from(data))
             }
-            ColumnData::String(data, _) => Arc::new(data.to_arrow()),
+            ColumnData::String(data, _) => Arc::new(data.to_arrow(Some(nulls))),
```
Comment (alamb) [inline mutable_batch/src/column.rs:301]: this was the problem: the null mask that was computed didn't make it to the created `StringArray`

File: query_tests/src/influxrpc/field_columns.rs [hunk 2]
```diff
@@ -165,7 +165,7 @@ async fn test_field_name_plan() {
             Field {
                 name: "field2".into(),
                 data_type: DataType::Utf8,
-                last_timestamp: 1000, // Need to verify with alamb if 1000 is the right one. It looks to me it should be 100
```
Comment (alamb) [inline query_tests/src/influxrpc/field_columns.rs:168]: LOL @NGA-TRAN  had spotted this problem many 🌔 's ago

File: mutable_batch_lp/src/lib.rs [hunk 3]
```diff
@@ -225,7 +225,83 @@ mod tests {
...
+
+        // Verify the nullness of the string column ("" not the same as null)
+        let f0 = &batch.columns()[0];
+        assert!(f0.is_valid(0));
+        assert!(!f0.is_valid(1));
+
+        // Verify the nullness of the f1 column ("" not the same as null)
+        let f1 = &batch.columns()[1];
+        assert!(!f1.is_valid(0));
+        assert!(f1.is_valid(1));
```
Comment (NGA-TRAN) [inline mutable_batch_lp/src/lib.rs:262]: 👍

File: query_tests/src/influxrpc/field_columns.rs (hunk 2 above)
Comment (NGA-TRAN) [reply query_tests/src/influxrpc/field_columns.rs:168]: 🎉

**Review by NGA-TRAN — APPROVED** [review]
This NULL never-ended issues remind me of easy-to-us vs more-work-for-developer

**Review by tustvold — APPROVED** [review]
Nice find :+1: 

Perhaps we should have an option to format nulls differently for use by tests...

**Comment by alamb** [comment]
> Perhaps we should have an option to format nulls differently for use by tests...

Might definitely be nice

**Pull request merged**